nvflare/apis/impl/wf_comm_server.py

```diff
@@ -135,7 +139,7 @@ def _set_stats(self, fl_ctx: FLContext):
                     "collector must be an instance of GroupInfoCollector, but got {}".format(type(collector))
                 )
             collector.add_info(
-                group_name=self.controller._name,
+                group_name=self.name,
```

Comment:
Here `self.name='WFCommServer'` while `self.controller._name='ScatterAndGather'`. This change is fine, but I do remember one of the integration tests uses the stats collector and expects the Controller name (eg 'ScatterAndGather'), so that would have to be changed accordingly.

Reply:
The reason I changed this is to avoid accessing private "_name" of self.controller. Besides, the group_name can be anything. If you know where the test is, please modify it - it shouldn't be checking the name.

Reply:
Sure, I will look into which tests have to be changed